src/main/java/reposense/report/SummaryJson.java

```diff
@@ -30,7 +31,7 @@ public class SummaryJson {
 
     public SummaryJson(List<RepoConfiguration> repos, ReportConfiguration reportConfig, String reportGeneratedTime,
             LocalDateTime sinceDate, LocalDateTime untilDate, boolean isSinceDateProvided, boolean isUntilDateProvided,
-            String repoSenseVersion, Set<Map<String, String>> errorSet, String reportGenerationTime, String zoneId) {
+            String repoSenseVersion, Set<Map<String, String>> errorSet, String reportGenerationTime, ZoneId zoneId) {
```

Comment:
Since you mentioned that the system tests would break if `ZoneId` type were used in `SummaryJson`, it's okay to leave the input type as a `String` since the `zoneId` is unlikely to be converted to another type once it has been passed into `SummaryJson`.

Reply:
My idea is that if we allow `String` input, then we should validate it in the constructor? I believe you also mentioned this in the issue.

My proposed solution is therefore to directly accept a `ZoneId` object as input and hence the validation part is somewhat handled within `ZoneId`. The downside is probably what you mentioned below, that it kinda out of place compared to others.

Would you still like me to change it back to the original version? If so, what about the validation part?

Reply:
@yhtMinceraft1010X What do you think about this?

Reply:
Switched back to `String zoneId`

Reply:
I would like to revisit this, I think I agree with @SkyBlaise99 here, it would be a better design to ensure that the ZoneId object is passed into the constructor here, rather than the String value. The `zoneId` variable in the `SummaryJson` object should also change type to ZoneId.

Can I check what errors you are getting with the systemtests? Is it potentially because we are lacking a serializer to convert the ZoneId object to a string and vice versa?

Reply:
The error in `systemtest` should be due to the change in json file output.

Previously the json file is as such:
``` json
{
    "reportGeneratedTime":"Mon Jul 11 11:31:42 2022 SGT",
    "reportGenerationTime":" 23.02 second(s)",
    "zoneId":"Asia/Singapore",
    "rest of the data":"..."
}
```

After using `ZondId` object type:
``` json
    {"reportGeneratedTime":"Mon Jul 11 11:35:16 2022 SGT",
    "reportGenerationTime":" 22.39 second(s)",
    "zoneId":{"id":"Asia/Singapore"},
    "rest of the data":"..."
}
```

There is now an extra bracket there which I'm not sure if it will affect frontend side.

Reply:
I see, I think it is still preferable that we take in a ZoneId object, as well as store it as a ZoneId. The json file is printed slightly differently as we have not defined a custom serializer. We currently do that for `LocalDateTime` and `FileType`:

https://github.com/reposense/RepoSense/blob/40d87be077f56fdfdc7b1ca40b56449b3a6bbff0/src/main/java/reposense/util/FileUtil.java#L104-L108

You can refer to the way the serializer is implemented in the `FileType` class, and I believe it should be possible to resolve the system test failures.